lib/fdk/listener.rb

```diff
@@ -46,23 +51,41 @@ def listen(&block)
 
     def handle_request(fn_block:)
       local_socket = socket.accept
-      req = WEBrick::HTTPRequest.new(WEBrick::Config::HTTP)
-      resp = WEBrick::HTTPResponse.new(WEBrick::Config::HTTP)
+      req, resp = new_req_resp
       req.parse(local_socket)
       FDK.debug "got request #{req}"
+      log_frame_header(req.header)
       fn_block.call(req, resp)
       resp["Connection"] = "close" # we're not using keep alives sadly
       resp.send_response(local_socket)
       FDK.debug "sending resp  #{resp.status}, #{resp.header}"
       local_socket.close
     end
 
+    def new_req_resp
+      req = WEBrick::HTTPRequest.new(WEBrick::Config::HTTP)
+      resp = WEBrick::HTTPResponse.new(WEBrick::Config::HTTP)
+      [req, resp]
+    end
+
     def socket_path
       @socket_path ||= url[5..url.length]
     end
 
     def private_socket_path
       socket_path + ".private"
     end
+
+    def log_frame_header(headers)
+      return if @fn_logframe_name.nil? || @fn_logframe_hdr.nil?
+
+      k = @fn_logframe_hdr.downcase
+      v = headers[k]
+      return if v.nil?
```

Comment:
maybe `.empty?` instead of nil here, since it's possibly an empty string?

Reply:
empty list or empty string I suppose

Reply:
Fixed. Have to do a nil check first anyway unfortunately else you get `Error in request handling undefined method `empty?' for nil:NilClass`